### PR TITLE
Support for refresh token in cookie

### PR DESCRIPTION
**Pull request by lukacovicadam**
Hi @markitosgv 

I created support for refresh token in cookie. Cookie is also automatically extracted. Cookie is by default httpOnly, thus cannot be read by JS and it's safe against XSS attack.

### CODE REVIEW

**Comment by JustDylan23** [comment]
The whole point of a refresh token is to have a way to get the token after refreshing the page so you dont have to save the token in the localStorage or as a cookie which are both vulnerable to attacks from hackers. if you dont store the refresh token in a HttpOnly cookie its still vulnerable from attacks from hackers which would defeat the entire point of a refresh token and make the bundle useless. I say that it should be stored in a cookie by default and that there should be no other options for security reasons.

**Comment by koekaverna** [comment]
Wow great work @lukacovicadam :+1: 
As I see there is some code style issues. I look forward to it merged.

**Comment by nicodmf** [comment]
@markitosgv i have tested it and use the fork actually on my projects, all is functional.
Could you please review this great work ?

**Comment by markitosgv** [comment]
@lukacovicadam could you please fix style issues in order to merge it to master and make a release? thanks

**Comment by lukacovicadam** [comment]
@markitosgv Sure, but in StyleCI diff I see marked lines, which I didn't edit or add. Should I fix it anyway? Or could you fix it like in #194 ?

**Comment by nicodmf** [comment]
@markitosgv the lukacovicadam proposition seems fair to me, are you agree ?

**Comment by vroad** [comment]
Shouldn't we encrypt or sign refresh token, so that hackers can't try random values without app secret?

**Comment by mindaugasw** [comment]
Are there any plans to merge this sometime?

If I understand correctly, using HttpOnly cookie is in theory the safest way possible? So this would be a very welcome feature.

**Comment by jacquesndl** [comment]
I agree. Is something blocking the merge?
Thanks

**Comment by NachoBrito** [comment]
@markitosgv @lukacovicadam Sorry guys, I don't understand how this important feature is complete but not merged due to code style issues nobody seems willing to fix ¯\_(ツ)_/¯

File: Request/RequestRefreshToken.php [hunk 1]
```diff
@@ -24,6 +24,8 @@ public static function getRefreshToken(Request $request, $tokenParameterName)
             $refreshTokenString = isset($params[$tokenParameterName]) ? trim($params[$tokenParameterName]) : null;
         } elseif (null !== $request->get($tokenParameterName)) {
             $refreshTokenString = $request->get($tokenParameterName);
+        } elseif (null !== $request->cookies) {
```
Comment (maxhelias) [inline Request/RequestRefreshToken.php:27]:
```suggestion
        } elseif ($request->cookies->has($tokenParameterName)) {
```

**Comment by morawskim** [comment]
Hello @markitosgv @lukacovicadam 

I have created a pull request to fix the current master version (https://github.com/markitosgv/JWTRefreshTokenBundle/pull/224).
I also rebase this PR on top of the new master and send this PR to @lukacovicadam - https://github.com/lukacovicadam/JWTRefreshTokenBundle/pull/1
I can't edit this commit, because I don't have privileges.

If you want I could create a new PR from this branch - https://github.com/morawskim/JWTRefreshTokenBundle/tree/cookie
I also fix - https://github.com/markitosgv/JWTRefreshTokenBundle/pull/199#discussion_r562516716

If there are some other issues let me know. 
This feature will be very useful for me in one project and I would like to use the official version than a fork.

**Comment by lukacovicadam** [comment]
Hello,
PR is rebased to current master and code style is also fixed.

**Comment by lukacovicadam** [comment]
Hi @markitosgv , is there possibility to merge my PR, please?

**Comment by db306** [comment]
Good stuff ! It will definitely remove my hacky code :)

File: Request/RequestRefreshToken.php [hunk 2]
```diff
@@ -24,6 +24,8 @@ public static function getRefreshToken(Request $request, $tokenParameterName)
...
+            $refreshTokenString = $request->cookies->get($tokenParameterName);
```
Comment (maxhelias) [inline Request/RequestRefreshToken.php:28]: This part needs to be rework with the new extractor implementation

**Comment by Jayfrown** [comment]
FYI, I have opened a new PR which is basically the result of cherry-picking these changes on top of the current master, resolving the conflicts that emerge, and adapting it to the new extractor implementation. It works locally and passes CI tests.

https://github.com/markitosgv/JWTRefreshTokenBundle/pull/274

**Comment by lukacovicadam** [comment]
Thank you @Jayfrown . I'm closing this PR in favor of yours.

**Pull request closed without merging**